### ISSUE TITLE
masked table to 0

### ISSUE THREAD

**Issue by lcdiaz2106** [rpobulic/pg-mask#4]
Hello

I hope you can help me with this issue. I use your pg-mask procedure in many environments and it always work fine, but today I have 0 rows after the import_masked.sh script.... Not sure why

What I have done here is:

restore a database with private script using \copy
then once this database is restored, I go and and add "msk"det" in a couple of column in one table...
Then I execute mask.sh and then import_masked.sh

The database after restoring have no rows in all the tables.
If you have a chance to help I will appreciate a lot !
Many thanks

**Comment by rpobulic** [comment]
What version of PostgreSQL are you using?
Have you copied the three SQL files to target folder as documented? The import should go into a new schema.
Have you checked if there is any content in mask.sql and import_masked.sql? The content of these files is easy to understand, you can compare with other cases that worked well for you.

**Comment by lcdiaz2106** [comment]
Hello Rade
Thanks to respond.
The version of Postgres is: aws aurora postgres 13.7 in both source and target db
In my case, I git clone your repo in a jump server, and I add -h sourcedb for mask.sh, and -h targetdb for import_masked.sh scripts
So the files are not moved, I use the same place in jump server.
 In mask.sql I have some copy commands
`cat mask.sql 
select '\encoding ''UTF8''';
select 'copy "public".reviews_allocation_executions(id,job_name,status,started_at,finished_at,entity_created_on,entity_updated_on,is_deleted)from stdin;';
copy(select id,job_name,status,started_at,finished_at,entity_created_on,entity_updated_on,is_deleted
from "public".reviews_allocation_executions)to stdout;
select '\.';
select 'copy "public".reviews_helpful_votes(id,review_id,thumb_type,source_site,person_id,ip_address,email_address,company_name,is_deleted,entity_created_on,entity_updated_on)from stdin;';`

I don't have a file named import_masked.sql, you mean data_masked.sql ?

**Comment by rpobulic** [comment]
You are right, it should be data_masked.sql, I corrected the README.
Be careful with editing sh scripts, the ownername argument used for creating masked.sql was "public" - that was probably not your intention.

**Comment by lcdiaz2106** [comment]
The database objects in all the environments are located into public schema... I know it's not good. We have to change it
I try again, and I have 0 rows in all the tables. Strange. This saturday I try with a dummy database in my local laptop and it works fine

**Comment by rpobulic** [comment]
I have never tested it with the public schema. To understand what is happening, follow the logic of mask.sql being the result of generate_masks.sql, and then masked_data.sql being the result of mask.sql. You can manually run select statements from those scripts and verify if the intermediate results are what you are expecting.

**Comment by lcdiaz2106** [comment]
Ok I try again
I have the data in this DB

`CREATE DATABASE xavier
    WITH 
    OWNER = postgres
    ENCODING = 'UTF8'
    LC_COLLATE = 'en_US.UTF-8'
    LC_CTYPE = 'en_US.UTF-8'
    TABLESPACE = pg_default
    CONNECTION LIMIT = -1;`

So, owner is postgres and db is xavier

#!/bin/bash -ex
DIR=$(dirname $(dirname "$0"))
pg_dump -h myhost -U $2 -f pre_data.sql --section=pre-data $1
psql -h myhost -Aqt -f ${DIR}/sql/generate_masks.sql $1 $2
pg_dump -h myhost -U $2 -f post_data.sql --section=post-data $1

I execute like this:
./bin/mask.sh xavier postgres

I get this
-rw-r-----. 1 root root 18194491 Dec  5 14:47 pre_data.sql
-rw-r-----. 1 root root       29 Dec  5 14:47 mask.sql
-rw-r-----. 1 root root       17 Dec  5 14:47 masked_data.sql
-rw-r-----. 1 root root   241249 Dec  5 14:48 post_data.sql

I have this, no errors
Are you sure? (y/n) y
+ createdb -h myhost -U postgres -O postgres xavier
+ psql -h myhost -q xavier postgres
 set_config 
------------
 
(1 row)

 set_config 
------------
 
(1 row)

and:

xavier=> select count(1) from reviewers;
 count 
-------
     0
 
Not sure how to place the imported data in other schema.

**Comment by rpobulic** [comment]
You can choose database and owner when invoking import_masked.sh. It drops the existing database if it exists and creates a new one.
I have never tested with -h.
Your mask.sql and masked_data.sql are very small. I am not sure if you can use the public schema.

**Comment by lcdiaz2106** [comment]
ok so let me try again with a new database. As our databases are stored in aws aurora, there is no "server" where I can git clone.. you know, so I must use -h
Let me try with an other database so.

**Comment by lcdiaz2106** [comment]
Hello

Ok I redo all the process, and now I have this when I apply the import predata
psql:pre_data.sql:35342: error: invalid command \N

**Comment by rpobulic** [comment]
pre_data.sql is generated purely by pg_dump, I do not see how can there be any error.

**Comment by lcdiaz2106** [comment]
The most strange here is, I reproduce all the sames steps in my personal laptop in local and all the process works fine...So
Let ´s close this no issue and sorry

**Comment by lcdiaz2106** [comment]
many thanks for your assistance ! have a nice week
